src/components/datasets/DatasetDataTypes.js

```diff
@@ -0,0 +1,128 @@
+import { Button, Col, Row, Table, Typography } from "antd";
+
+import PropTypes from "prop-types";
+import { datasetPropTypesShape, projectPropTypesShape } from "../../propTypes";
+import { clearDatasetDataType } from "../../modules/metadata/actions";
+import { fetchDatasetDataTypesSummaryIfPossible } from "../../modules/datasets/actions";
+import genericConfirm from "../ConfirmationModal";
+import { nop } from "../../utils/misc";
+import DataTypeSummaryModal from "./datatype/DataTypeSummaryModal";
+
+const NA_TEXT = <span style={{ color: "#999", fontStyle: "italic" }}>N/A</span>;
+
+const DatasetDataTypes = React.memo(({isPrivate, project, dataset, onDatasetIngest:
+    onDatasetIngest, isFetchingDatasets }) => {
+    const dispatch = useDispatch();
+
+    const datasetDataTypes = useSelector((state) => state.datasetDataTypes.itemsById[dataset.identifier]);
+    const datasetSummaries = useSelector((state) => state.datasetSummaries.itemsById[dataset.identifier]);
+
+    const [datatypeSummaryVisible, setDatatypeSummaryVisible] = useState(false);
+    const [selectedDataType, setSelectedDataType] = useState(null);
+
+    const selectedSummary = (selectedDataType !== null && datasetSummaries)
+        ? datasetSummaries[selectedDataType.id]
+        : {};
+
+    const handleDeleteDataType = async (dataType) => {
+        genericConfirm({
+            title: `Are you sure you want to delete the "${dataType.label || ""}" data type?`,
+            content: "Deleting this means all instances of this data type contained in the dataset " +
+                "will be deleted permanently, and will no longer be available for exploration.",
+            onOk: async () => {
+                await dispatch(clearDatasetDataType(dataset.identifier, dataType.id));
+                await dispatch(fetchDatasetDataTypesSummaryIfPossible(dataset.identifier));
+            },
+        });
+    };
+
+    const showDatatypeSummary = (dataType) => {
+        setSelectedDataType(dataType);
+        setDatatypeSummaryVisible(true);
+    };
+
+    const dataTypesColumns = [
+        {
+            title: "Name",
+            key: "label",
+            render: (dt) =>
+                isPrivate ? (
+                    <a style={{ fontFamily: "monospace" }} onClick={() => showDatatypeSummary(dt)}>
+                        {dt.label ?? NA_TEXT}
+                    </a>
+                ) : dt.label ?? NA_TEXT,
+            defaultSortOrder: "ascend",
+            sorter: (a, b) => a.label.localeCompare(b.label),
+        },
+        {
+            title: "Count",
+            dataIndex: "count",
+            render: (c) => (c ?? NA_TEXT),
+        },
+        ...(isPrivate ? [
+            {
+                title: "Actions",
+                key: "actions",
+                width: 230,
+                render: (dt) => (
+                    <Row gutter={10}>
+                        <Col span={12}>
+                            <Button
```

Comment:
this button should be disabled or not present if one cannot ingest into this data type - either based on workflows, or some flag set on the data type list in Katsu + Gohan - what do you think is best?

Reply:
also, if i click this, is project/dataset/data type correctly pre-selected?